test/FisherPrice_Alpaka/particleProcessor.h

```diff
@@ -0,0 +1,145 @@
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * @file particleProcessor.h
+ * @brief simulates electrons and photons - it can calculate ionisaton energy loss, make a particle undergo brehmstrahhlung or pair production and move a particle along the x-direction. 
+ * All class functions are specified to run the accelerator (device) only.
+ * @author Davide Costanzo (d.costanzo@sheffield.ac.uk) and Mark Hodgkinson (d.costanzo@sheffield.ac.uk)
+ * */
+
+#ifndef PARTICLEPROCESSOR_H
+#define PARTICLEPROCESSOR_H
+
+#include "part.h"
+#include "particleStack.h"
+#include "sensitive.h"
+#include <alpaka/alpaka.hpp>
+
+class particleProcessor {
+
+  public:
+    /** @brief A minimal constructor */
+    ALPAKA_FN_ACC particleProcessor (){}
```

Comment:
Besides "abstracting" the apaka-dependent headers, we may want also to abstract macros like ALPAKA_FN_ACC to avoid depending on specific back-ends

Reply:
@agheta, this is good point and something we can/should address through Issue #24?